### PR TITLE
feat: explore state management

### PR DESCRIPTION
**Pull request by kayleegeorge**
Keep track of current filters w/ state management

Screen Shot 2022-07-22 at 12 13 20 PM

### CODE REVIEW

File: src/components/Explore/TokenRow.tsx [hunk 1]
```diff
@@ -17,20 +18,26 @@ import {
...
+  filterTimeAtom,
+  sortCategoryAtom,
+  sortDirectionAtom,
+  useSortCategory,
+  useToggleFavorite,
+} from './state'
 import { TIME_DISPLAYS } from './TimeSelector'
 
-enum Category {
+export enum Category {
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:31]: lets make a new file structure:
```

- Explore
    - TokenTable
        - index.tsx
        - types.tsx
        - Row.tsx
    -  TokenDetails
        - index.tsx
        - ...
    - state.ts
    - types.ts
```

Then move these shared types into the right places. If it's Table specific put it in that subfolder, else you can put in the top level `types.ts`.

File: src/components/Explore/TokenRow.tsx [hunk 2]
```diff
@@ -318,21 +323,22 @@ function getHeaderDisplay(category: string, timeframe: string): string {
...
-  category: string // TODO: change this to make it work for trans
-  sortDirection: SortDirection
+  category: Category // TODO: change this to make it work for trans
   isSorted: boolean
   sortable: boolean
   timeframe: string
 }) {
   const theme = useTheme()
+  const sortDirection = useAtomValue<SortDirection>(sortDirectionAtom)
+  const handleSortCategory = useSortCategory(category)
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:337]: `useSetSortCategory`

File: src/components/Explore/TokenTable/TokenRow.tsx [hunk 3]
```diff
@@ -402,35 +400,31 @@ export function HeaderRow({ timeframe }: { timeframe: string }) {
       price={
         <HeaderCell
           category={Category.price}
-          sortDirection={SortDirection.Decreasing}
-          isSorted={sortedBy === Category.price}
+          isSorted={sortCategory === Category.price}
```
Comment (kayleegeorge) [inline src/components/Explore/TokenTable/TokenRow.tsx:403]: move this logic to HeaderCell

File: src/components/Explore/state.ts [hunk 4]
```diff
@@ -21,3 +30,23 @@ export function useToggleFavorite(tokenAddress: string) {
...
+
+/* keep track of sort category for token table */
+export function useSetSortCategory(category: Category) {
+  const [sortCategory, setSortCategory] = useAtom(sortCategoryAtom)
+  const [sortDirection, setDirectionCategory] = useAtom(sortDirectionAtom)
+
+  return useCallback(() => {
+    const oppositeDirection =
+      sortDirection === SortDirection.Increasing ? SortDirection.Decreasing : SortDirection.Increasing
+    console.log('categor: ' + category)
```
Comment (vm) [inline src/components/Explore/state.ts:42]: rm logs

File: src/components/Explore/TokenTable/TokenRow.tsx [hunk 5]
```diff
@@ -402,35 +400,31 @@ export function HeaderRow({ timeframe }: { timeframe: string }) {
...
           sortable
           timeframe={timeframe}
         />
       }
       marketCap={
         <HeaderCell
           category={Category.market_cap}
-          sortDirection={SortDirection.Decreasing}
-          isSorted={sortedBy === Category.market_cap}
+          isSorted={sortCategory === Category.market_cap}
```
Comment (kayleegeorge) [inline src/components/Explore/TokenTable/TokenRow.tsx:419]: change to marketCap

File: src/components/Explore/state.ts [hunk 6]
```diff
@@ -21,3 +30,23 @@ export function useToggleFavorite(tokenAddress: string) {
...
   }, [favoriteTokens, tokenAddress, updateFavoriteTokens])
 }
+
+/* keep track of sort category for token table */
+export function useSetSortCategory(category: Category) {
+  const [sortCategory, setSortCategory] = useAtom(sortCategoryAtom)
+  const [sortDirection, setDirectionCategory] = useAtom(sortDirectionAtom)
+
+  return useCallback(() => {
+    const oppositeDirection =
```
Comment (vm) [inline src/components/Explore/state.ts:40]: move this into the equals case

**Review by vm — APPROVED** [review]
lgtm after comments

File: src/pages/Explore/index.tsx [hunk 7]
```diff
@@ -52,6 +56,17 @@ const FiltersWrapper = styled.div`
 `
 
 const Explore = () => {
+  const resetFilterString = useResetAtom(filterStringAtom)
```
Comment (kayleegeorge) [inline src/pages/Explore/index.tsx:59]: @vm change here

Comment (vm) [reply src/pages/Explore/index.tsx:59]: explain IRL

**Pull request merged**